Index: Makeconfig
===================================================================
--- Makeconfig	2008-12-18 23:34:16.000000000 -0500
+++ Makeconfig	2008-12-18 23:34:52.000000000 -0500
@@ -574,9 +574,9 @@
 
 # Extra flags to pass to GCC.
 ifeq ($(all-warnings),yes)
-+gccwarn := -Wall -Wwrite-strings -Winline -Wcast-qual -Wbad-function-cast -Wmissing-noreturn -Wmissing-prototypes -Wmissing-declarations -Wcomment -Wcomments -Wtrigraphs -Wsign-compare -Wfloat-equal -Wmultichar
++gccwarn := -Wall -Wwrite-strings -Wcast-qual -Wbad-function-cast -Wmissing-noreturn -Wmissing-prototypes -Wmissing-declarations -Wcomment -Wcomments -Wtrigraphs -Wsign-compare -Wfloat-equal -Wmultichar
 else
-+gccwarn := -Wall -Wwrite-strings -Winline
++gccwarn := -Wall -Wwrite-strings
 endif
 +gccwarn-c = -Wstrict-prototypes
 
Index: debug/readlink_chk.c
===================================================================
--- debug/readlink_chk.c	2008-12-18 23:37:31.000000000 -0500
+++ debug/readlink_chk.c	2008-12-18 23:38:02.000000000 -0500
@@ -25,7 +25,7 @@
 
 
 ssize_t
-__readlink_chk (const char *path, void *buf, size_t len, size_t buflen)
+__readlink_chk (const char *path, char *buf, size_t len, size_t buflen)
 {
   if (len > buflen)
     __chk_fail ();
Index: debug/readlinkat_chk.c
===================================================================
--- debug/readlinkat_chk.c	2008-12-18 23:37:31.000000000 -0500
+++ debug/readlinkat_chk.c	2008-12-18 23:37:48.000000000 -0500
@@ -21,7 +21,7 @@
 
 
 ssize_t
-__readlinkat_chk (int fd, const char *path, void *buf, size_t len,
+__readlinkat_chk (int fd, const char *path, char *buf, size_t len,
 		  size_t buflen)
 {
   if (len > buflen)

