--- linux/fs/compat_ioctl.c	2008-02-25 19:20:20.000000000 -0500
+++ linux-2.6.24.3/fs/compat_ioctl.c	2008-03-09 13:21:12.000000000 -0400
@@ -113,6 +113,8 @@
 
 #include <linux/sort.h>

+#include <linux/msdos_fs.h>
+
 #ifdef CONFIG_SPARC
 #include <asm/fbio.h>
 #endif
@@ -1504,6 +1504,8 @@
 
 IGNORE_IOCTL(VFAT_IOCTL_READDIR_BOTH32)
 IGNORE_IOCTL(VFAT_IOCTL_READDIR_SHORT32)
+COMPATIBLE_IOCTL(FAT_IOCTL_GET_ATTRIBUTES)
+COMPATIBLE_IOCTL(FAT_IOCTL_SET_ATTRIBUTES)
 
 #ifdef CONFIG_SPARC
 /* Sparc framebuffers, handled in sbusfb_compat_ioctl() */
