--- linux/include/linux/time.h	2008-12-18 20:23:44.000000000 -0500
+++ linux-2.6.24.3/include/linux/time.h	2008-12-18 20:24:02.000000000 -0500
@@ -169,7 +169,7 @@
  * @a:		pointer to timespec to be incremented
  * @ns:		unsigned nanoseconds value to be added
  */
-static inline void timespec_add_ns(struct timespec *a, u64 ns)
+static inline void timespec_add_ns(struct timespec *a, volatile u64 ns)
 {
 	ns += a->tv_nsec;
 	while(unlikely(ns >= NSEC_PER_SEC)) {
